tools/csvtk/collapse.xml

```diff
@@ -0,0 +1,198 @@
+    </macros>
+    <expand macro="requirements" />
+    <command detect_errors="exit_code"><![CDATA[
+
+###################
+## Start Command ##
+###################
+
+csvtk collapse --num-cpus "\${GALAXY_SLOTS:-1}"
+
+    ## Add additional flags as specified ##
+    #######################################
+    $ignore_case
+    $global_param.illegal_rows
+    $global_param.empty_rows
+    $global_param.header
+    $global_param.lazy_quotes
+
+    ## Set Tabular input/output flag if input is tabular ##
+    #######################################################
+    #if $in_1.is_of_type("tabular"):
+        -t -T
+    #end if
+
+    ## Set Input ##
+    ###############
+    '$in_1'
+
+    ## Specify fields for collapse ##
+    #################################
+    -F -f '$column_text.in_text'
+
+    ## other ##
+    ###########
+    -v '$value_column.in_text'
+    -s '$separator'
+
+    ## To output ##
+    ###############
+    > collapsed
+
+    ]]></command>
+    <inputs>
+        <expand macro="singular_input" />
+        <conditional name="column_text" >
+            <param type="select" name="select" label="Select key column(s) based on" argument="-F -f">
+                <option value="string">Column Name(s)</option>
+                <option value="column">Column Number(s)</option>
+            </param>
+            <when value="column">
+                <param type="data_column" name="in_text"
+                        data_ref="in_1"
+                        multiple="true"
+                        label="Input column number(s)"
+                        help="Select column(s) to use as keys for collapse"
+                        />
+            </when>
+            <when value="string">
+                <param type="text" name="in_text"
+                    optional="false"
+                    label="Input column name(s)"
+                    help="Specify columns to be used as keys for collapse. 
+                    Multiple columns can be given if separated by a ' , '. 
+                    Column numbers can be given too - ex. '1,2' will target columns 1 and 2. 
+                    Please see the help section below for more detailed info"
+                    />
+            </when>
+        </conditional> 
+        <conditional name="value_column" >
+            <param type="select" name="select" label="Select value column to collapse based on" argument="-v">
+                <option value="string">Column Name</option>
+                <option value="column">Column Number</option>
+            </param>
+            <when value="column">
+                <param type="data_column" name="in_text"
+                        data_ref="in_1"
+                        multiple="false"
+                        label="Input column number"
+                        help="Select column to collapse on"
+                        />
+            </when>
+            <when value="string">
+                <param type="text" name="in_text"
```

Comment:
That could be replaced by the stuff that you already have in your macros, left? A sanitizer would also be cool here.

Reply:
Thanks! Added the sanitizer to all of the text inputs